### PR TITLE
Added relativistic discriminator loss used in ESRGAN paper.

### PR DESCRIPTION
**Pull request by bnb32**
This changes the previous disc loss calc (and adversarial loss) to use the relativistic versions described in the ESRGAN paper. The symmetry enables us to remove the adversarial loss function, and instead just swap the arguments in the new disc loss calc, and also seems worth doing since this improved on the SRGAN framework.

### CODE REVIEW

**Comment by grantbuster** [comment]
@bnb32 Starting review - to clarify, you only implemented the feature "Relativistic average GAN (RaGAN) [20], which
learns to judge “whether one image is more realistic than the other” rather than “whether one image is real or fake”" from the Wang paper, right? Seems like Wang did a lot of different things and I'm trying to track what to pay attention to. 

Also, does this completely revise the previous disc method or is it an option?

File: sup3r/models/base.py [hunk 1]
```diff
@@ -904,6 +814,84 @@ def train(
...
+            loss = loss_gen
+        elif train_disc:
+            loss = loss_disc
+
+        loss_details = {
+            'loss_gen': loss_gen,
+            'loss_gen_content': loss_gen_content,
+            'loss_gen_advers': loss_gen_advers,
+            'loss_disc': loss_disc,
+        }
```
Comment (grantbuster) [inline sup3r/models/base.py:890]:
```suggestion
        loss_details = {}

        if train_gen:
            loss_gen_content, loss_gen_content_details = (
                self.calc_loss_gen_content(hi_res_true, hi_res_gen)
            )
            loss_gen_advers = self.calc_loss_disc(
                disc_out_true=disc_out_gen, disc_out_gen=disc_out_true
            )
            loss = loss_gen_content + weight_gen_advers * loss_gen_advers
            loss_details['loss_gen'] = loss
            loss_details['loss_gen_content'] = loss_gen_content,
            loss_details['loss_gen_advers'] = loss_gen_advers,

        elif train_disc:
            loss = self.calc_loss_disc(
                disc_out_true=disc_out_true, disc_out_gen=disc_out_gen
            )
            loss_details['loss_disc'] = loss

```

**Review by grantbuster — APPROVED** [review]
Minor suggestion but LGTM

File: sup3r/pipeline/forward_pass.py [hunk 2]
```diff
@@ -386,7 +386,9 @@ def get_node_cmd(cls, config):
     @classmethod
     def _constant_output_check(cls, out_data, allowed_const):
         """Check if forward pass output is constant. This can happen when the
-        chunk going through the forward pass is too big.
+        chunk going through the forward pass is too big. This is due to a
+        tensorflow padding bug, with the padding mode set to 'reflect'.
+        https://github.com/tensorflow/tensorflow/issues/91027
```
Comment (grantbuster) [inline sup3r/pipeline/forward_pass.py:391]: You might note that in our current TF version 2.15 this actually just results in scrambled outputs.

File: sup3r/models/base.py [hunk 3]
```diff
@@ -904,6 +814,84 @@ def train(
...
+        loss_gen = loss_gen_content + weight_gen_advers * loss_gen_advers
+        loss_disc = self.calc_loss_disc(
+            disc_out_true=disc_out_true, disc_out_gen=disc_out_gen
+        )
+
+        loss = None
+        if train_gen:
+            loss = loss_gen
+        elif train_disc:
+            loss = loss_disc
```
[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/models/base.py:883]: Good call. We've had this inefficient setup for a while.

[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/models/base.py:883]: yeah no blame just reading our old code and thinking well that could have been done better haha

[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/models/base.py]: @grantbuster Actually, we need to compute the disc loss every batch to track whether to train it or not. We can skip the gen loss calcs though.

[the remark this answers is not in the document]
Comment (bnb32) [reply sup3r/models/base.py]: Nvm, how's this solution? https://github.com/NREL/sup3r/pull/261/commits/d122aceedc26e29664038bb49c532b87900fc9fb

[the remark this answers is not in the document]
Comment (grantbuster) [reply sup3r/models/base.py]: LGTM
